### ISSUE TITLE
[FEATURE] GUI for all commands

### ISSUE THREAD

**Issue by megargayu** [booksaw/BetterTeams#132]
## GUI for all commands
This means "buttons" in "chests" - for example, clicking an item in a chest that pops up on your screen will execute a certain command. This will take a **long** time to implement, so don't expect it any time soon. Definitely not a v4 feature.

On `/team gui`, you can see a main GUI that has all these features:
- [ ] Warps
  - [ ] Warp descriptions ("lore") **[or not]**
  - [ ] Warp items **[or not]** (this is the item used in the GUI that the player clicks to warp)
  - [ ] Home warp
- [ ] Team controls
  - [ ] Change team name/color/description/open
  - [ ] View team bans/unban player
  - [ ] See neutral and ally teams (and change their status - toggleable)
  - [ ] Balance/deposit/withdraw
- [ ] Players
  - [ ] View team players (click on their head as an item)
    - [ ] Demote/promote selected player
    - [ ] Kick/ban selected player
  - [ ] Invite player (paste command into player's chatbox, and ask for username)
- [ ] Other teams
  - [ ] See other teams (mimic `/team info`)
  - [ ] Join a team (if its open)
  - [ ] See invites 
- [ ] Change chat (ally/team/global)
- [ ] Leave team

Basically, making a GUI interface for all commands. Again, I don't expect this to be implemented soon (or maybe this idea would be even rejected because of the amount of work this would need).

**Comment by GuazP** [comment]
@megargayu this is managable if you make your own script using denizen/skript :)

**Comment by megargayu** [comment]
@GuazP Well, that is true, but it would be nice if it was inbuilt into the plugin as an integration. I was thinking of using booksaw's [Gui Api](https://github.com/booksaw/Gui-API) since it was made for GUIs (and it's made by booksaw), but I'm still deciding.

**Comment by booksaw** [comment]
In theory it would not be too complicated to with guiAPI

**Comment by megargayu** [comment]
@booksaw Yeah, and it could just be an integration so it wouldn't need to refactor that much code. When I have more time I will definitely dive in and see if I can code this using Gui API

**Comment by GuazP** [comment]
@megargayu Hey, just wonder, did you make any progress? Because I wonder if I should make workaround gui with denizen ;d
